apps/web/src/hooks/useAuth.ts

```diff
@@ -107,9 +107,27 @@ export function useAuth() {
     navigate(ROUTES.AUTH_LOGIN);
   }, [navigate, queryClient, segment]);
 
+  const redirectTo = useCallback(({ url, redirectURL }: { url: string; redirectURL?: string }) => {
+    const finalURL = new URL(url, window.location.origin);
+
+    if (redirectURL) {
+      finalURL.searchParams.append('redirect_url', redirectURL);
+    }
+
+    // Note: Do not use react-router-dom. The version we have doesn't do instant cross origin redirects.
+    window.location.href = finalURL.href;
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** Again here -- do we want to skip the history here by using `replace`?‏